site/src/sphinx/advanced-structured-logging.rst

```diff
@@ -148,6 +152,114 @@ automatically for you:
+
+    HttpClientBuilder cb = new HttpClientBuilder();
+    ...
+    cb.contentPreview(100);
+
+Note that the ``contentPreview()`` method enables the previews only for textual content
+which meets one of the following cases:
+
+- when its type matches ``text/*`` or ``application/x-www-form-urlencoded``.
+- when its charset has been specified. e.g. application/json; charset=utf-8.
+- when its subtype is ``xml`` or ``json``. e.g. application/xml, application/json.
+- when its subtype ends with ``+xml`` or ``+json``. e.g. application/atom+xml, application/hal+json
+
+You can also customize the previews by specifying your own :api:`ContentPreviewerFactory` implementation.
+The following example enables the textual preview of first 100 characters for the content type of ``text/*``,
+and the hex dump preview of first 100 bytes for other types:
+
+.. code-block:: java
+
+    import io.netty.buffer.ByteBufUtil;
+    import com.linecorp.armeria.common.MediaType;
+    import com.linecorp.armeria.common.logging.ContentPreviewer;
+
+    ServerBuilder sb = new ServerBuilder();
+
+    sb.contentPreviewerFactory((ctx, headers) -> {
+        MediaType contentType = headers.contentType();
+        if (contentType != null && contentType.is(MediaType.ANY_TEXT_TYPE)) {
+            // Produces the textual preview of the first 100 characters.
+            return ContentPreviewer.ofText(100);
+        }
+        // Produces the hex dump of the first 100 bytes.
+        return ContentPreviewer.ofBinary(100, byteBuf -> {
+            // byteBuf has no more than 100 bytes.
+            return ByteBufUtil.hexDump(byteBuf);
+        });
+    });
+
+You can write your own :api:`ContentPreviewer` to change the way to make the preview, e.g.
+
+.. code-block:: java
+
+    class HexDumpContentPreviewer implements ContentPreviewer {
+        private final StringBuilder builder = new StringBuilder();
+        @Nullable;
+        private String preview;
+
+        @Override
+        public void onHeaders(HttpHeaders headers) {
+            // Invoked when headers of a request or response is received.
+        }
+
+        @Override
+        public void onData(HttpData data) {
+            // Invoked when a new content is received.
+            builder.append(ByteBufUtil.hexDump(data.array(), data.offset(), data.length()));
+        }
+
+        @Override
+        public boolean isDone() {
+            // If it returns true, no further event is invoked but produce().
+            return preview != null;
+        }
+
+        @Override
+        public String produce() {
+            // Invoked when a request or response ends.
+            if (preview != null) {
+                return preview;
+            }
+            preview = builder.build();
+            builder = null;
+            return preview;
+        }
+    }
+    ...
+    ServerBuilder sb = new ServerBuilder();
+    ...
+    sb.contentPreviewFactory((ctx, headers) -> {
+        return new HexDumpContentPreviewer();
```

Comment:
Could be simplified into:

    sb.contentPreviewFactory((ctx, headers) -> new HexDumpContentPreviewer());

I didn't try but I'm curious if this works: (Never mind if it doesn't work :-))

    sb.contentPreviewFactory(HexDumpContentPreviewer::new);